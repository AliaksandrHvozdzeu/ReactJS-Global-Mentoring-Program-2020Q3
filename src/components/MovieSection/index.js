import MovieSection from './js/MovieSection';

export default MovieSection;
