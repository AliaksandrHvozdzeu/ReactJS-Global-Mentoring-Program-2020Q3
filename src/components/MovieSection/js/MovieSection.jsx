import React from "react";
import Wrapper from "../../Wrapper";
import ErrorBoundary from "../../ErrorBoundary";
import Filter from "../../MovieFilter";
import MovieLoadMessage from "../../MovieLoadMessage";
import "../css/MovieSection.css";

export default function MovieSection() {

  const SearchResults = React.lazy(() => import("../../SearchResults"));

  const loadWindow = (
    <MovieLoadMessage />
  );

  return (
    <>
      <main>
        <ErrorBoundary>
          <Wrapper>
            <Filter />
            <React.Suspense fallback={loadWindow}>
              <SearchResults />
            </React.Suspense>
          </Wrapper>
        </ErrorBoundary>
      </main>
    </>
  );

}
