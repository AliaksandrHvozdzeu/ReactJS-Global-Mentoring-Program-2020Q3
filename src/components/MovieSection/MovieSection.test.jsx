import React from 'react';
import rerender from 'react-test-renderer';
import MovieSection from './js/MovieSection';

describe('when MovieSection', () => {

  test('then snapshot created', () => {
    const component = rerender.create(<MovieSection/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
