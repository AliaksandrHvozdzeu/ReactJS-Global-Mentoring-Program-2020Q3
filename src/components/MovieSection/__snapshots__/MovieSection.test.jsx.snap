// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieSection then snapshot created 1`] = `
<main>
  <div
    className="error"
  >
    Something went wrong. Please try again later.
  </div>
</main>
`;
